include/seqan3/io/alignment_file/record.hpp

```diff
@@ -0,0 +1,415 @@
+    decltype(auto) sequence_position() const &
+    {
+        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The (pairwise) alignment stored in an seqan3::alignment object.
+    decltype(auto) alignment() &&
+    {
+        return get_impl(field_constant<seqan3::field::alignment>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::alignment
+    decltype(auto) alignment() const &&
+    {
+        return get_impl(field_constant<seqan3::field::alignment>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::alignment
+    decltype(auto) alignment() &
+    {
+        return get_impl(field_constant<seqan3::field::alignment>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::alignment
+    decltype(auto) alignment() const &
+    {
+        return get_impl(field_constant<seqan3::field::alignment>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The identifier of the (reference) sequence that SEQ was aligned to.
+    decltype(auto) reference_id() &&
+    {
+        return get_impl(field_constant<seqan3::field::ref_id>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_id
+    decltype(auto) reference_id() const &&
+    {
+        return get_impl(field_constant<seqan3::field::ref_id>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_id
+    decltype(auto) reference_id() &
+    {
+        return get_impl(field_constant<seqan3::field::ref_id>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_id
+    decltype(auto) reference_id() const &
+    {
+        return get_impl(field_constant<seqan3::field::ref_id>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The (reference) "sequence" information, usually a range of nucleotides or amino acids.
+    decltype(auto) reference_sequence() &&
+    {
+        return get_impl(field_constant<seqan3::field::ref_seq>{}, static_cast<tuple_base_t &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_sequence
+    decltype(auto) reference_sequence() const &&
+    {
+        return get_impl(field_constant<seqan3::field::ref_seq>{}, static_cast<tuple_base_t const &&>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_sequence
+    decltype(auto) reference_sequence() &
+    {
+        return get_impl(field_constant<seqan3::field::ref_seq>{}, static_cast<tuple_base_t &>(*this));
+    }
+    //!\copydoc seqan3::sam_record::reference_sequence
+    decltype(auto) reference_sequence() const &
+    {
+        return get_impl(field_constant<seqan3::field::ref_seq>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief Sequence (REF_SEQ) relative start position (0-based), unsigned value.
```

Comment:
SAM does not have `REF_SEQ`

Reply:
```suggestion
    //!\brief Sequence (seqan3::sam_record::reference_sequence) relative start position (0-based), unsigned value.
```

Reply:
you are right and we don't set it currently in sam. You can infer this from the id-sequence-map.